spark/ingestion/src/test/scala/feast/ingestion/StreamingPipelineIT.scala

```diff
@@ -136,16 +136,91 @@ class StreamingPipelineIT extends SparkSpec with ForAllTestContainer {
     query.processAllAvailable()
 
     rows.foreach { r =>
-      val storedValues = jedis.hgetAll(encodeEntityKey(r, config.featureTable)).asScala.toMap
+      val encodedEntityKey = encodeEntityKey(r, config.featureTable)
+      val storedValues     = jedis.hgetAll(encodedEntityKey).asScala.toMap
       storedValues should beStoredRow(
         Map(
           featureKeyEncoder("unique_drivers") -> r.getUniqueDrivers,
-          "_ts:driver-fs"                     -> new java.sql.Timestamp(r.getEventTimestamp.getSeconds * 1000)
+          "_ts:driver-fs"                     -> new java.sql.Timestamp(r.getEventTimestamp.getSeconds * 1000),
+          "_ex:driver-fs"                     -> new java.sql.Timestamp(Timestamps.MAX_VALUE.getSeconds * 1000)
         )
       )
+      val keyTTL = jedis.ttl(encodedEntityKey).toInt
+      keyTTL shouldEqual -1
     }
   }
 
+  "Streaming pipeline" should "store messages from kafka to redis with expiry time equal to the largest of (event_timestamp + max_age) for all feature " +
+    "tables associated with the entity" in new Scope {
+      val maxAge = 86400
+      val configWithMaxAge = config.copy(
+        source = kafkaSource,
+        featureTable = config.featureTable.copy(maxAge = Some(maxAge))
+      )
+      val query = StreamingPipeline.createPipeline(sparkSession, configWithMaxAge).get
+      query.processAllAvailable() // to init kafka consumer
+
+      val rows = generateDistinctRows(rowGenerator, 100, groupByEntity)
+
+      val ingestionTimeUnix = System.currentTimeMillis()
+      rows.foreach(sendToKafka(kafkaSource.topic, _))
+
+      query.processAllAvailable()
+
+      rows.foreach { r =>
+        val encodedEntityKey = encodeEntityKey(r, config.featureTable)
+        val storedValues     = jedis.hgetAll(encodedEntityKey).asScala.toMap
+        storedValues should beStoredRow(
+          Map(
+            featureKeyEncoder("unique_drivers") -> r.getUniqueDrivers,
+            "_ts:driver-fs"                     -> new java.sql.Timestamp(r.getEventTimestamp.getSeconds * 1000),
+            "_ex:driver-fs" -> new java.sql.Timestamp(
+              (r.getEventTimestamp.getSeconds + maxAge) * 1000
+            )
+          )
+        )
+        val keyTTL = jedis.ttl(encodedEntityKey).toLong
+        keyTTL should (be <= (r.getEventTimestamp.getSeconds + maxAge - ingestionTimeUnix / 1000) and be > 0L)
+      }
+
+      val increasedMaxAge               = 86400 * 2
```

Comment:
Can we have test for max age = 0 instead, since this flow was already tested with Batch Pipeline?